src/migration/mod.rs

```diff
@@ -0,0 +1,341 @@
+use self::strategy::{Strategy, UpdatedIndyItem};
+use crate::crypto::generic_array::typenum::U32;
+use crate::protect::hmac_key::HmacKey;
+use crate::protect::ProfileKey;
+use crate::Error;
+use askar_crypto::alg::chacha20::{Chacha20Key, C20P};
+use askar_crypto::kdf::argon2::{Argon2, PARAMS_MODERATE};
+use askar_crypto::kdf::KeyDerivation;
+use askar_crypto::repr::KeySecretBytes;
+use sha2::Sha256;
+use sqlx::sqlite::SqliteRow;
+use sqlx::{Connection, Row, SqliteConnection};
+
+mod strategy;
+
+const CHACHAPOLY_KEY_LEN: u8 = 32;
+const CHACHAPOLY_NONCE_LEN: u8 = 12;
+
+#[derive(Serialize, Deserialize, Debug, Default)]
+pub(crate) struct IndyKeyMetadata {
+    keys: Vec<u8>,
+    master_key_salt: Vec<u8>,
+}
+
+/// Copies: https://github.com/hyperledger/indy-sdk/blob/83547c4c01162f6323cf138f8b071da2e15f0c90/libindy/indy-wallet/src/wallet.rs#L18
+#[derive(Serialize, Deserialize)]
+pub(crate) struct IndyKey {
+    type_key: Chacha20Key<C20P>,
+    name_key: Chacha20Key<C20P>,
+    value_key: Chacha20Key<C20P>,
+    item_hmac_key: HmacKey<Sha256, U32>,
+    tag_name_key: Chacha20Key<C20P>,
+    tag_value_key: Chacha20Key<C20P>,
+    tag_hmac_key: HmacKey<Sha256, U32>,
+}
+
+pub(crate) struct IndyKeyWithMasterAndSalt {
+    indy_key: IndyKey,
+    master: Chacha20Key<C20P>,
+    salt: Vec<u8>,
+}
+
+pub(crate) struct IndySdkToAriesAskarMigration {
+    wallet_key: String,
+    wallet_name: String,
+    conn: SqliteConnection,
+}
+
+impl IndySdkToAriesAskarMigration {
+    pub async fn new(spec_uri: &str, wallet_name: &str, wallet_key: &str) -> Result<Self, Error> {
```

Comment:
We probably need to add support for the keyDerivationMethod here?

Reply:
Yeah I had that in the back of my head the whole time.  Let me check if the python script has some logic for this.

Reply:
If you can use the Askar key derivcation logic for this, we have this mapping in AFJ. Left is indy, right is Askar

```
  const correspondenceTable = {
    [KeyDerivationMethod.Raw]: StoreKeyMethod.Raw,
    [KeyDerivationMethod.Argon2IInt]: `${StoreKeyMethod.Kdf}:argon2i:int`,
    [KeyDerivationMethod.Argon2IMod]: `${StoreKeyMethod.Kdf}:argon2i:mod`,
  }
```

Reply:
```
export enum KeyDerivationMethod {
  /** default value in indy-sdk. Will be used when no value is provided */
  Argon2IMod = 'ARGON2I_MOD',
  /** less secure, but faster */
  Argon2IInt = 'ARGON2I_INT',
  /** raw wallet master key */
  Raw = 'RAW',
}
```

Reply:
```
export declare enum StoreKeyMethod {
    Raw = "raw",
    Kdf = "kdf",
    None = "none"
}
```

Reply:
Oh thanks! Yeah I try my best to use askar existing methods. I got some working now with custom kdf, but I will transform it now using askar existing functionality.

Reply:
Latest commit adds argon2i MOD and INT and also RAW. 

The tests are also included now and use the RAW method to make it a lot quicker.